tests/unit/expressjs/test_app.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""ExpressJS charm unit tests for the generic app module."""
+
+import pathlib
+
+import pytest
+
+from paas_charm.app import App, WorkloadConfig
+from paas_charm.charm_state import CharmState, IntegrationsState
+from paas_charm.expressjs.charm import ExpressJSConfig
+
+
+@pytest.mark.parametrize(
+    "set_env, user_defined_config, framework_config, integrations, expected",
+    [
+        pytest.param(
+            {},
+            {},
+            {},
+            None,
+            {
+                "PORT": "8080",
+                "NODE_ENV": "production",
+                "APP_SECRET_KEY": "foobar",
+                "APP_OTHERCONFIG": "othervalue",
+                "APP_BASE_URL": "https://paas.example.com",
+            },
+        ),
```

Comment:
Nit: Adding test ID to make it easier to locate :D
```suggestion
        pytest.param(
            {},
            {},
            {},
            None,
            {
                "PORT": "8080",
                "NODE_ENV": "production",
                "APP_SECRET_KEY": "foobar",
                "APP_OTHERCONFIG": "othervalue",
                "APP_BASE_URL": "https://paas.example.com",
            },
            id="minimal environments"
        ),
```